src/core/components/bitswap.js

```diff
@@ -4,21 +4,34 @@ const OFFLINE_ERROR = require('../utils').OFFLINE_ERROR
 const promisify = require('promisify-es6')
 const setImmediate = require('async/setImmediate')
 const Big = require('big.js')
+const PeerId = require('peer-id')
 
 function formatWantlist (list) {
   return Array.from(list).map((e) => e[1])
 }
 
 module.exports = function bitswap (self) {
   return {
-    wantlist: () => {
+    wantlist: promisify((peerId, callback) => {
+      if (!callback) {
+        callback = peerId
+        peerId = undefined
+      }
+
       if (!self.isOnline()) {
-        throw new Error(OFFLINE_ERROR)
+        return setImmediate(() => callback(new Error(OFFLINE_ERROR)))
       }
 
-      const list = self._bitswap.getWantlist()
-      return formatWantlist(list)
-    },
+      let list
+      if (peerId) {
+        peerId = PeerId.createFromB58String(peerId)
```

Comment:
A try/catch around this will allow us to callback with an error for invalid user input.